src/main/java/com/radixdlt/crypto/encryption/PrivateKeyEncrypter.java

```diff
@@ -0,0 +1,169 @@
+/*
+ *
+ *  * (C) Copyright 2020 Radix DLT Ltd
+ *  *
+ *  * Radix DLT Ltd licenses this file to you under the Apache License,
+ *  * Version 2.0 (the "License"); you may not use this file except in
+ *  * compliance with the License.  You may obtain a copy of the
+ *  * License at
+ *  *
+ *  *  http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing,
+ *  * software distributed under the License is distributed on an
+ *  * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ *  * either express or implied.  See the License for the specific
+ *  * language governing permissions and limitations under the License.
+ *
+ */
+
+package com.radixdlt.crypto.encryption;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import java.io.IOException;
+import java.io.Reader;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.security.GeneralSecurityException;
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.KeySpec;
+import java.util.Arrays;
+import javax.crypto.Cipher;
+import javax.crypto.SecretKey;
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.PBEKeySpec;
+import javax.crypto.spec.SecretKeySpec;
+
+import com.google.gson.stream.JsonReader;
+import com.radixdlt.crypto.ECKeyPair;
+import com.radixdlt.crypto.ECKeyUtils;
+import com.radixdlt.crypto.Hash;
+import com.radixdlt.crypto.MacMismatchException;
+import com.radixdlt.crypto.keystore.Cipherparams;
+import com.radixdlt.crypto.keystore.Crypto;
+import com.radixdlt.crypto.keystore.Keystore;
+import com.radixdlt.crypto.keystore.Pbkdfparams;
+import com.radixdlt.utils.Bytes;
+
+public final class PrivateKeyEncrypter {
+
+    private static final int ITERATIONS = 100000;
+    private static final int KEY_LENGTH = 32;
+    private static final String DIGEST = "sha512";
+    private static final String ALGORITHM = "aes-256-ctr";
+
+    private PrivateKeyEncrypter() { }
+
+    public static String createEncryptedPrivateKey(String password) throws GeneralSecurityException {
+        ECKeyPair ecKeyPair = new ECKeyPair();
```

Comment:
Could use null check

Reply:
Actually, all methods here would benefit from a null check

Reply:
@flotothemoon Hmm like this:

```java
    public static String createEncryptedPrivateKey(String password) throws GeneralSecurityException {
        ECKeyPair ecKeyPair = new ECKeyPair();
        String privateKey = Objects.requireNonNull(Bytes.toHexString(ecKeyPair.getPrivateKey()));
        byte[] salt = getSalt().getBytes(StandardCharsets.UTF_8);

        SecretKey derivedKey = Objects.requireNonNull(getSecretKey(password, salt, ITERATIONS, KEY_LENGTH));

        Cipher cipher = Objects.requireNonNull(Cipher.getInstance("AES/CTR/NoPadding"));
        cipher.init(Cipher.ENCRYPT_MODE, derivedKey);
        byte[] iv = cipher.getParameters().getParameterSpec(IvParameterSpec.class).getIV();

        String cipherText = encrypt(cipher, privateKey);
        byte[] mac = generateMac(derivedKey.getEncoded(), Bytes.fromHexString(cipherText));

        Keystore keystore = Objects.requireNonNull(createKeystore(ecKeyPair, cipherText, mac, iv, salt));

        Gson gson = new GsonBuilder().setPrettyPrinting().create();
        return Objects.requireNonNull(gson.toJson(keystore));
    }
```

Reply:
Not that's not what I meant. 
I meant checking the input arguments to the method (password)

Reply:
ah, roger that!